src/core/api.c

```diff
@@ -616,6 +621,204 @@ mender_api_publish_deployment_status(const char *id, mender_deployment_status_t
         ret = MENDER_FAIL;
     }
 
+END:
+
+    /* Release memory */
+    mender_free(response);
+    mender_free(path);
+    mender_free(payload);
+    cJSON_Delete(json_payload);
+
+#ifdef CONFIG_MENDER_DEPLOYMENT_LOGS
+    /* Do this after we have released memory above, potentially giving us some
+       extra room we may need. */
+    if ((MENDER_OK == ret) && (MENDER_DEPLOYMENT_STATUS_FAILURE == deployment_status)) {
+        /* Successfully reported a deployment failure, upload deployment
+           logs.  */
+        if (MENDER_OK != mender_api_publish_deployment_logs(id)) {
+            mender_log_error("Failed to publish deployment logs");
+        }
+    }
+#endif /* CONFIG_MENDER_DEPLOYMENT_LOGS */
+
+    return ret;
+}
+
+#ifdef CONFIG_MENDER_DEPLOYMENT_LOGS
+static void
+append_depl_log_msg(char *msg, void *ctx) {
```

Comment:
Why `void *`? There is only one caller and no magic needed I think.

Reply:
Because that's the type of the visitor function.